### PR TITLE
vlc: add srt support

### PR DESCRIPTION
**Pull request by oxalica**
###### Motivation for this change

This enables the protocol `srt://` for vlc and downstream packages.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
```
4 packages marked as broken and skipped:
libsForQt512.elisa libsForQt512.phonon-backend-vlc libsForQt514.elisa tribler

1 package failed to build:
obs-studio-plugins.obs-ndi # Depends on non-distributable file and need manual `nix-prefetch-url`.

26 packages built:
arcan.arcan arcan.arcan-wrapped arcan.durden-wrapped arcan.everyone-wrapped arcan.pipeworld-wrapped arcan.prio-wrapped arcan.xarcan elisa libsForQt5.phonon-backend-vlc libsForQt514.phonon-backend-vlc libvlc megaglest minitube obs-studio obs-studio-plugins.looking-glass-obs obs-studio-plugins.obs-gstreamer obs-studio-plugins.obs-move-transition obs-studio-plugins.obs-multi-rtmp obs-studio-plugins.wlrobs openlpFull pympress python38Packages.python-vlc python39Packages.python-vlc reaper strawberry vlc
```
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131866` at b39c2d0e run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>obs-studio-plugins.obs-ndi</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/7d65f283aebb56b085d25a1c81394adc">libsForQt514.elisa</a></li>
    <li><a href="https://gist.github.com/05ccb8b3782155c10d8f8f9345df5a86">tribler</a></li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>libsForQt512.elisa</li>
    <li>strawberry</li>
  </ul>
</details>
<details>
  <summary>14 packages built successfully:</summary>
  <ul>
    <li>arcan.arcan</li>
    <li>arcan.xarcan</li>
    <li>libsForQt5.elisa (libsForQt515.elisa ,plasma5Packages.elisa)</li>
    <li>libsForQt5.phonon-backend-vlc (libsForQt515.phonon-backend-vlc ,plasma5Packages.phonon-backend-vlc)</li>
    <li>libsForQt512.phonon-backend-vlc</li>
    <li>libsForQt514.phonon-backend-vlc</li>
    <li>libvlc</li>
    <li>megaglest</li>
    <li>minitube</li>
    <li>pympress</li>
    <li>python38Packages.python-vlc</li>
    <li>python39Packages.python-vlc</li>
    <li>reaper</li>
    <li>vlc (vlc_qt5)</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/applications/video/vlc/default.nix [hunk 1]
```diff
@@ -14,6 +14,7 @@
 , skins2Support ? !onlyLibVLC, freetype
 , removeReferencesTo
 , chromecastSupport ? true, protobuf, libmicrodns
+, srtSupport ? true, srt
```
Comment (SuperSandro2000) [inline pkgs/applications/video/vlc/default.nix:17]: srt is pretty small. We can remove this flag.

**Comment by erdnaxe** [comment]
LGTM although my review is weak as I'm a beginner in Nix.
Enabling SRT support in VLC is nice-to-have, especially for people that needs an equivalent to RTMP which has limitations.

---

Result of `nixpkgs-review pr 131866` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.elisa</li>
    <li>libsForQt512.phonon-backend-vlc</li>
    <li>libsForQt514.elisa</li>
    <li>tribler</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>obs-studio-plugins.obs-ndi</li>
  </ul>
</details>
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>arcan.arcan</li>
    <li>arcan.arcan-wrapped</li>
    <li>arcan.durden-wrapped</li>
    <li>arcan.everyone-wrapped</li>
    <li>arcan.pipeworld-wrapped</li>
    <li>arcan.prio-wrapped</li>
    <li>arcan.xarcan</li>
    <li>libsForQt5.elisa (libsForQt515.elisa ,plasma5Packages.elisa)</li>
    <li>libsForQt5.phonon-backend-vlc (libsForQt515.phonon-backend-vlc ,plasma5Packages.phonon-backend-vlc)</li>
    <li>libsForQt514.phonon-backend-vlc</li>
    <li>libvlc</li>
    <li>megaglest</li>
    <li>minitube</li>
    <li>obs-studio</li>
    <li>obs-studio-plugins.looking-glass-obs</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>obs-studio-plugins.obs-move-transition</li>
    <li>obs-studio-plugins.obs-multi-rtmp</li>
    <li>obs-studio-plugins.wlrobs</li>
    <li>pympress</li>
    <li>python38Packages.python-vlc</li>
    <li>python39Packages.python-vlc</li>
    <li>reaper</li>
    <li>strawberry</li>
    <li>vlc (vlc_qt5)</li>
  </ul>
</details>

File: pkgs/applications/video/vlc/default.nix (hunk 1 above)
Comment (fpletz) [reply pkgs/applications/video/vlc/default.nix]: Also, SRT is becoming more and more popular and is a solid replacement for RTMP. Thanks, I would've also suggested this.

**Pull request merged**